SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -440,26 +398,70 @@ void InterestFactory::AddServerSelfInterest(Interest& OutInterest) const
 	// Add a query for the load balancing worker (whoever is delegated the ACL) to read the authority intent
 	Query LoadBalanceQuery;
 	LoadBalanceQuery.Constraint.EntityIdConstraint = EntityId;
-	LoadBalanceQuery.ResultComponentId = TArray<Worker_ComponentId>{ SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID };
+	LoadBalanceQuery.ResultComponentId = ResultType{ SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID };
 	AddComponentQueryPairToInterestComponent(OutInterest, SpatialConstants::ENTITY_ACL_COMPONENT_ID, LoadBalanceQuery);
 }
 
-void InterestFactory::AddComponentQueryPairToInterestComponent(Interest& OutInterest, const Worker_ComponentId ComponentId, const Query& QueryToAdd)
+void InterestFactory::AddSystemQuery(Interest& OutInterest, const QueryConstraint& LevelConstraint) const
 {
-	if (!OutInterest.ComponentInterestMap.Contains(ComponentId))
+	const USpatialGDKSettings* Settings = GetDefault<USpatialGDKSettings>();
+
+	QueryConstraint CheckoutRadiusConstraint = CreateCheckoutRadiusConstraints();
+	QueryConstraint AlwaysInterestedConstraint = CreateAlwaysInterestedConstraint();
+	QueryConstraint AlwaysRelevantConstraint = CreateAlwaysRelevantConstraint();
+
+	QueryConstraint SystemDefinedConstraints;
+
+	if (CheckoutRadiusConstraint.IsValid())
 	{
-		ComponentInterest NewComponentInterest;
-		OutInterest.ComponentInterestMap.Add(ComponentId, NewComponentInterest);
+		SystemDefinedConstraints.OrConstraint.Add(CheckoutRadiusConstraint);
+	}
+
+	if (AlwaysInterestedConstraint.IsValid())
+	{
+		SystemDefinedConstraints.OrConstraint.Add(AlwaysInterestedConstraint);
+	}
+
+	if (AlwaysRelevantConstraint.IsValid())
+	{
+		SystemDefinedConstraints.OrConstraint.Add(AlwaysRelevantConstraint);
+	}
+
+	// Add the level constraint here as all client queries need to make sure they don't check out anything outside their loaded levels.
+	QueryConstraint SystemAndLevelConstraint;
+	SystemAndLevelConstraint.AndConstraint.Add(SystemDefinedConstraints);
+	SystemAndLevelConstraint.AndConstraint.Add(LevelConstraint);
+
+	Query SystemQuery;
```

Comment:
ClientSystemQuery?

Reply:
well, it's just the system query. The system query was defined before this PR. The server just makes a subset of the system query, fondly nicknamed the `ServerSystemQuery`. If you'd rather this wasn't the case, can change.